src/sentry/static/sentry/app/views/projectReleaseTracking.jsx

```diff
@@ -0,0 +1,273 @@
+import React from 'react';
+import _ from 'underscore';
+
+import ApiMixin from '../mixins/apiMixin';
+import {t} from '../locale';
+import PluginList from '../components/pluginList';
+import LoadingError from '../components/loadingError';
+import LoadingIndicator from '../components/loadingIndicator';
+
+const ProjectReleaseTracking = React.createClass({
+  propTypes: {
+    organization: React.PropTypes.object,
+    project: React.PropTypes.object
+  },
+
+  mixins: [ApiMixin],
+
+  getInitialState() {
+    return {
+      loading: true,
+      error: false,
+      pluginList: [],
+      webhookUrl: '',
+      token: ''
+    };
+  },
+
+  componentDidMount() {
+    this.fetchData();
+  },
+
+  fetchData() {
+    let {orgId, projectId} = this.props.params;
+    let done = _.after(2, () => {
+      this.setState({loading: false});
+    });
+    this.api.request(`/projects/${orgId}/${projectId}/releases/token/`, {
+      method: 'GET',
+      success: data =>
+        this.setState({
+          webhookUrl: data.webhookUrl,
+          token: data.token
+        }),
+      error: () => {
+        this.setState({
+          error: true
+        });
+      },
+      complete: done
+    });
+    this.getPluginConfig(done);
+  },
+
+  getPluginConfig(done) {
+    let {orgId, projectId} = this.props.params;
+    this.api.request(`/projects/${orgId}/${projectId}/plugins/`, {
+      success: data => {
+        this.setState({
+          pluginList: data.filter(p => p.type === 'release-tracking')
+        });
+      },
+      error: () => {
+        this.setState({
+          error: true
+        });
+      },
+      complete: done
+    });
+  },
+
+  onEnablePlugin(plugin) {
+    this.setState({
+      pluginList: this.state.pluginList.map(p => {
+        if (p.id !== plugin.id) return p;
+        return {
+          ...plugin,
+          enabled: true
+        };
+      })
+    });
+    this.getPluginConfig();
+  },
+
+  onDisablePlugin(plugin) {
+    this.setState({
+      pluginList: this.state.pluginList.map(p => {
+        if (p.id !== plugin.id) return p;
+        return {
+          ...plugin,
+          enabled: false
+        };
+      })
+    });
+  },
+
+  onSubmit(evt) {
+    evt.preventDefault();
+    this.regenerateToken();
+  },
+
+  regenerateToken() {
+    let {orgId, projectId} = this.props.params;
+    this.api.request(`/projects/${orgId}/${projectId}/releases/token/`, {
+      method: 'POST',
+      data: {project: projectId},
+      success: data => {
+        this.setState({
+          token: data.token,
+          webhookUrl: data.webhookUrl
+        });
+        this.getPluginConfig();
```

Comment:
can we also add a success message here with the `Your deploy token has been regenerated. You will need to update any pre-existing deploy hooks.` messaging?